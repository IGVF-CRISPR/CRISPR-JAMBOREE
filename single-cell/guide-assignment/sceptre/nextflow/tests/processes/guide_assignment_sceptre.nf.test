nextflow_process {

    name "Test Process guide_assignment_sceptre"
    script "processes/guide_assignment_sceptre.nf"
    process "guide_assignment_sceptre"

    test("guide_assignment_sceptre on Gasperini data") {

        when {
            params {
            }
            process {
                """
                input[0] = file("${projectDir}/data/gasperini_guide_assignment_input.h5mu")
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process).match() }
            )
        }
    }

    test("guide_assignment_sceptre on Papalexi data") {

        when {
            params {
            }
            process {
                """
                input[0] = file("${projectDir}/data/papalexi_guide_assignment_input.h5mu")
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process).match() }
            )
        }
    }
}