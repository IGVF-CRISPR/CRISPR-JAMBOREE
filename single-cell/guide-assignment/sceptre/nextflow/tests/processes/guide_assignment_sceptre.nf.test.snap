{
    "guide_assignment_sceptre on Gasperini data": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,55b5053a0d9e82d3ff90e4f00738e0c4"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "guide_assignment_sceptre",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T17:10:30.843387"
    },
    "guide_assignment_sceptre on Papalexi data": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": null,
                "exitStatus": 0,
                "out": {
                    "0": [
                        "mudata_out.h5mu:md5,00ccc18f103a33a6ffeaf40d292dbc00"
                    ]
                },
                "failed": false,
                "stdout": [
                    
                ],
                "errorMessage": null,
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 1,
                    "tasksSucceeded": 1
                },
                "name": "guide_assignment_sceptre",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-05-08T17:13:16.894199"
    }
}